cli/azd/pkg/project/framework_service.go

```diff
@@ -6,12 +6,20 @@ package project
 import (
 	"context"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools"
 )
 
 type FrameworkService interface {
-	RequiredExternalTools() []tools.ExternalTool
-	Package(ctx context.Context, progress chan<- string) (string, error)
-	InstallDependencies(ctx context.Context) error
-	Initialize(ctx context.Context) error
+	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	Initialize(ctx context.Context, serviceConfig *ServiceConfig) error
+	Restore(
+		ctx context.Context,
+		serviceConfig *ServiceConfig,
+	) *async.TaskWithProgress[*ServiceRestoreResult, ServiceProgress]
+	Build(
```

Comment:
I love the idea of supporting doing `build` in parallel!!  calling `azd build` and all your services are built at the same time!!!

On the same time, I am a little worried about that the UI would it be for that...  Right now, all `progress` handles would be competing for the same console, I think.

Reply:
There isn't anything stopping from using performing any of these in parallel but I'm not going to introduce that here.  Agreed that our UX would need an overall to figure out how we could implement UX progress reporting in that case.